engine/src/main/java/org/hibernate/search/query/dsl/sort/SortAdditionalSortFieldContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+import org.apache.lucene.search.SortField;
```

Comment:
@gunnarmorling I asked this on the JIRA ticket, and apparently this DSL is a temporary solution and will be reworked in 6.0, which is when we'll actually abstract from Lucene.

I think the point was that the generic solution is not hard to find, it's just hard to implement without adding APIs/SPIs in other places (`FullTextQuery` in particular) or even altering some APIs (`QueryBuilder` in particular is not well-suited to provide implementation-dependent sorts).